src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -755,59 +818,87 @@ private void ProcessError(Exception error)
             }
         }
 
-        private void Commit(IImmutableSet<TopicPartitionOffset> commitMap, Action<object> sendReply)
+        private void Commit(IImmutableSet<TopicPartitionOffset> commitMap, IImmutableSet<IActorRef> replyTo)
         {
+            var watch = Stopwatch.StartNew();
             try
             {
                 _commitRefreshing.UpdateRefreshDeadlines(commitMap.Select(tp => tp.TopicPartition)
                     .ToImmutableHashSet());
-
-                var watch = Stopwatch.StartNew();
+                _commitsInProgress += 1;
 
                 _consumer.Commit(commitMap);
-
+                _commitsInProgress -= 1;
                 watch.Stop();
                 if (watch.Elapsed >= _settings.CommitTimeWarning)
                     _log.Warning(
                         $"Kafka commit took longer than `commit-time-warning`: {watch.ElapsedMilliseconds} ms");
 
-                Self.Tell(new KafkaConsumerActorMetadata.Internal.Committed(commitMap));
-                sendReply(Akka.Done.Instance);
+                _commitRefreshing.Committed(commitMap);
+
+                foreach (var s in replyTo)
+                {
+                    s.Tell(Done.Instance);
+                }
             }
-            catch (Exception ex)
+            catch (TopicPartitionOffsetException offsetException)
             {
-                sendReply(new Status.Failure(ex));
+                watch.Stop();
+                if (offsetException.Error.Code == ErrorCode.RebalanceInProgress)
```

Comment:
If we had failed commits due to a `ErrorCode.RebalanceInProgress` we can retry those and they will succeed so long as the partitions they belong to weren't revoked from us. If they were revoked they'll be filtered out inside the partition assignment handler.

Reply:
I thought about inspecting each of the partition offsets for errors individually for this purpose too but decided to just send them all instead.